### PR TITLE
Added support for Gradle and finished login page

### PR DESCRIPTION
**Pull request by w2ji**
Using Gradle for managing dependency and initial facebook login page

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+apply plugin: 'android'
+
+dependencies {
+    compile fileTree(dir: 'libs', include: '*.jar')
+    compile 'com.facebook:facebook-android-sdk:+@aar'
+}
+
+android {
+    compileSdkVersion 17
```
Comment (divad12) [inline build.gradle]: Did we want to support ICS and above, so SDK version 15?

File: res/values/strings.xml [hunk 2]
```diff
@@ -1,4 +1,6 @@
 <?xml version="1.0" encoding="utf-8"?>
 <resources>
     <string name="app_name">flow_android</string>
+    <string name="app_id">289196947861602</string>
```
Comment (divad12) [inline res/values/strings.xml]: This is fine for now, but when we deploy to the Play Store, we'll want to change this to the prod app ID. We could try to detect whether we're running locally or on prod: http://stackoverflow.com/a/7428112/392426

Alternatively, we could just switch over to just using prod app ID once the mobile API gets built out more....

File: src/com/uwflow/flow_android/network/ApiRequests.java [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        params.put(Constants.FBID, facebookId);
+        params.put(Constants.FACEBOOK_ACCESS_TOKEN, facebookAccessToken);
+
+        FlowAsyncClient.post(Constants.API_LOGIN, params, new JsonHttpResponseHandler() {
+            @Override
+            public void onStart() {
+            }
+
+            @Override
+            public void onSuccess(int statusCode, Header[] headers, byte[] responseBody) {
```
Comment (divad12) [inline src/com/uwflow/flow_android/network/ApiRequests.java:22]: According to http://loopj.com/android-async-http/, it looks like we should be able to use [`PersistentCookieStore`](http://loopj.com/android-async-http/doc/com/loopj/android/http/PersistentCookieStore.html) to automatically handle all of the cookie-saving and passing logic, including saving to shared preferences.

File: src/com/uwflow/flow_android/network/FlowAsyncClient.java [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+    public static void post(String url, RequestParams params, AsyncHttpResponseHandler responseHandler) {
+        client.post(getAbsoluteUrl(url), params, responseHandler);
+    }
+
+    private static String getAbsoluteUrl(String relativeUrl) {
+        return Constants.BASE_URL + relativeUrl;
+    }
+
+    public static void storeCookie(Context c, String cookie) {
```
Comment (divad12) [inline src/com/uwflow/flow_android/network/FlowAsyncClient.java]: It seems to me from reading the docs page on http://loopj.com/android-async-http/ that if we call `client.setCookieStore()` in a static initializer, that should then automatically capture and store cookies from HTTP requests into the `PersistentCookieStore` that we then we give it.... I have not tested though.

File: build.gradle [hunk 5]
```diff
@@ -0,0 +1,50 @@
+buildscript {
```
Comment (divad12) [inline build.gradle:1]: Sweet, thanks for bringing this in. Much nicer than checking in dependencies into the codebase. :)

Is there anything we should do on our end to ensure we download the dependencies? If so, mind adding some instructions to the `README.md` file? :)

File: src/com/uwflow/flow_android/network/ApiRequests.java (hunk 3 above)
Comment (w2ji) [inline src/com/uwflow/flow_android/network/ApiRequests.java:22]: Yea, we are using PersistentCookieStore ATM

File: build.gradle (hunk 1 above)
Comment (w2ji) [inline build.gradle]: Yeah, I think version 14

File: res/values/strings.xml (hunk 2 above)
Comment (w2ji) [inline res/values/strings.xml]: I think there's a way to do it with environment variables. But I think this should be fine for the time being

File: build.gradle (hunk 1 above)
Comment (divad12) [inline build.gradle]: We're currently using version 14, but I think it's safe to use version 15 (which should be more reliable/less bugs than 14): http://stackoverflow.com/a/13544975/392426

File: src/com/uwflow/flow_android/constant/Constants.java [hunk 6]
```diff
@@ -0,0 +1,15 @@
+package com.uwflow.flow_android.constant;
+
+import java.util.HashMap;
+
+public class Constants {
+    public static final String BASE_URL = "https://divad12.fwd.wf/";
+    public static final String COOKIE = "flowCookie";
```
Comment (divad12) [inline src/com/uwflow/flow_android/constant/Constants.java:5]: Is this constant used?

File: res/values/strings.xml (hunk 2 above)
Comment (divad12) [inline res/values/strings.xml]: Also, can we call this `fb_app_id` just to namespace and clarify?

**Pull request merged**